### PR TITLE
Fix file name checking in rust handler

### PR DESCRIPTION
**Pull request by agatan**
In Rust handler, file name matching is not correct; the handler reports errors in `src/haserrors/mod.rs` while editing `src/noerrors/mod.rs`.

In the handler, the name of buffer and the name in the error message are striped using `fnamemodify(..., ':t')`, but they should be compared using suffix match.

### CODE REVIEW

File: autoload/ale/handlers/rust.vim [hunk 1]
```diff
@@ -50,7 +50,7 @@ function! ale#handlers#rust#HandleRustErrorsForFile(buffer, full_filename, lines
 
             if (
             \   l:span.is_primary
-            \   && (l:span_filename ==# l:filename || l:span_filename ==# '<anon>')
+            \   && (a:full_filename =~ (l:span.file_name . '$') || l:span_filename ==# '<anon>')
```
Comment (w0rp) [inline autoload/ale/handlers/rust.vim]: Could you try using `ale#path#IsBufferPath(a:buffer, l:span.file_name)` here, and let me know if that works? Matching with regex won't work in every case, as some other files might be similar enough that they match.

Comment (w0rp) [inline autoload/ale/handlers/rust.vim]: You may have to add `silent file ...` to the tests after changing this, so the tests run with the buffer set to a filename we should match against.

Comment (agatan) [inline autoload/ale/handlers/rust.vim]: Thank you for your fast reply!
`ale#path#IsBufferPath` does not work here. In my experience, `IsBufferPath`'s work is like this:

0. cwd = `/tmp/sample`, error file = `/tmp/sample/src/haserrors/mod.rs`, current file = `/tmp/sample/src/haserrors/mod.rs`
1. Get the file name of current buffer as `l:buffer_filename` ( = `/tmp/sample/src/haserrors/mod.rs`)
2. Apply `fnamemodify` to get enclosing directory of `l:buffer_filename` ( = `/tmp/sample/src/haserrors/` )
3. Concat the directory and error file name using `ale#path#GetAbsPath` ( = `/tmp/sample/src/haserrors/src/haserrors/mod.rs` )

`cargo` reports error with relative file path from the current directory, so the reported file name is `src/haserrors/mod.rs`.
(If you are in `src/haserrors`, the above solution works correctly)

Anyway, I will try to remove unused variables and add a test for the silent case.

**Comment by w0rp** [comment]
This will do for now. The regex matching of the filename isn't perfect, as certain characters in filenames will be treated as pattern characters, but this will be better that what's there at the moment.

I would like to work on writing a function which can solve this problem easily for any linter at some point. The same problem is being solved in different ways in different files.

Cheers! :beers:

**Pull request merged**